dockerfiles/puppeteer-context/mina_daemon_puppeteer.py

```diff
@@ -94,14 +94,14 @@ def stop_daemon():
 def inactive_loop():
   global active_daemon_request
 
-  with HTTPServer(('0.0.0.0', 3085), MockRequestHandler) as server:
-    while True:
-      server.handle_request()
-      signal.sigtimedwait(ALL_SIGNALS, 0)
-      if active_daemon_request:
-        start_daemon()
-        active_daemon_request = False
-        break
+  server = HTTPServer(('0.0.0.0', 3085), MockRequestHandler)
```

Comment:
My Python skills aren't great, but I think this will leak the httpserver -- Does the `with` not work with whatever version of python we switched to?

Instead we can put a try/finally and close it explicitly in the finally

Reply:
yeah, HTTPServer does not play nice with `with` in python 3.5, I'll try `try/finally` instead